analytics.py

```diff
@@ -0,0 +1,284 @@
+        list_means.append(newMean)
+
+    return list_means
+
+#++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+#
+# function find_crit_points
+#
+# Finds the critical points (min and max) of a list
+#++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+
+def find_crit_points(list_means):
+    entries = list_means
+    maxEntry = 0
+    minEntry = 2
+    for i in range(len(list_means)):
+        if entries[i] > maxEntry:
+            maxEntry = entries[i]
+        if entries[i] < minEntry:
+            minEntry = entries[i]
+
+    return minEntry,maxEntry
+
+#++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+#
+# function crit_point_check
+#
+# Checks if the observed value is within the range of the min and max
+#++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+
+
+def crit_point_check(minEntry, maxEntry, observed):
+    if observed < minEntry or observed > maxEntry:
+        return True
+    else:
+        return False
+
+
+def mean_center(points):
+    """
+    Given a set of points, compute the mean center
+
+    Parameters
+    ----------
+    points : list
+         A list of points in the form (x,y)
+
+    Returns
+    -------
+    x : float
+        Mean x coordinate
+
+    y : float
+        Mean y coordinate
+    """
+    x = 0
+    y = 0
+    numOfPoints = len(points)
+
+    for i in range(numOfPoints):
+        x = x + points[i][0]
+        y = y + points[i][1]
+
+    x = x/numOfPoints
+    y = y/numOfPoints
+    
+    return x, y
+
+def average_nearest_neighbor_distance(points):
+    """
+    Given a set of points, compute the average nearest neighbor.
+
+    Parameters
+    ----------
+    points : list
+             A list of points in the form (x,y)
+
+    Returns
+    -------
+    mean_d : float
+             Average nearest neighbor distance
+
+    References
+    ----------
+    Clark and Evan (1954 Distance to Nearest Neighbor as a
+     Measure of Spatial Relationships in Populations. Ecology. 35(4)
+     p. 445-453.
+    """
+    mean_d = 0
+    total = 0
+    local_nn = 0
+    numOfPoints = len(points)
+
+    for i in range(numOfPoints):
+        local_nn = 0 #reset local_nn for the new point
+        for j in range(numOfPoints):
+            if i != j:
+                newDistance = euclidean_distance(points[i],points[j])
+                if local_nn == 0:
+                    local_nn = newDistance
+                elif newDistance < local_nn:
+                    local_nn = newDistance
+
+        total = total + local_nn
+        
+    mean_d = total/numOfPoints
+        
+
+    return mean_d
+
+
+def minimum_bounding_rectangle(points):
+    """
+    Given a set of points, compute the minimum bounding rectangle.
+
+    Parameters
+    ----------
+    points : list
+             A list of points in the form (x,y)
+
+    Returns
+    -------
+     : list
+       Corners of the MBR in the form [xmin, ymin, xmax, ymax]
+    """
+    mbr = [0,0,0,0]
```

Comment:
Does this assume the MBR is always in a positive space, e.g. 0 - 360 degrees if a sphere?  What about -180 to 180 if we are working on a global dataset?